drivers/dma/dma_silabs_siwx91x.c

```diff
@@ -573,27 +573,39 @@ static int siwx91x_dma_init(const struct device *dev)
 	};
 	int ret;
 
-	ret = clock_control_on(cfg->clock_dev, cfg->clock_subsys);
-	if (ret) {
-		return ret;
-	}
+	if (action == PM_DEVICE_ACTION_RESUME) {
+		ret = clock_control_on(cfg->clock_dev, cfg->clock_subsys);
+		if (ret) {
+			return ret;
+		}
 
-	udma_handle = UDMAx_Initialize(&udma_resources, udma_resources.desc, NULL,
-				       (uint32_t *)&data->udma_handle);
-	if (udma_handle != &data->udma_handle) {
-		return -EINVAL;
-	}
+		udma_handle = UDMAx_Initialize(&udma_resources, udma_resources.desc, NULL,
+					       (uint32_t *)&data->udma_handle);
```

Comment:
Do you think `UDMAx_Initialize()` should be called on `PM_DEVICE_ACTION_RESUME` or on `PM_DEVICE_ACTION_TURN_ON`?

I believe Device runtime PM will only play with `PM_DEVICE_ACTION_SUSPEND`. `PM_DEVICE_ACTION_SUSPEND` may stop the clocks and the regulator but I believe it will retains the registers.

I assume System Managed PM won't retains the registers but it will call `PM_DEVICE_ACTION_TURN_OFF` and `PM_DEVICE_ACTION_TURN_ON`. So, `UDMAx_Initialize()` should be only done in `PM_DEVICE_ACTION_TURN_ON`.

@Martinhoff-maker, @asmellby, can you confirm my understanding?